core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.spec.ts

```diff
@@ -208,26 +208,44 @@ describe('Translation Modal Component', () => {
   });
 
   describe('when clicking on the translatable content', () => {
-    let target: HTMLElement;
+    const nonParagraphTarget: HTMLElement = document.createElement('div');
+    const mathTarget: HTMLElement = document.createElement(
+      'oppia-noninteractive-math');
+    let paragraphTarget: HTMLElement;
     let broadcastSpy: jasmine.Spy<(target: HTMLElement) => void>;
     let propagationSpy: jasmine.Spy<() => void>;
     beforeEach(fakeAsync(() => {
+      paragraphTarget = document.createElement('p');
       spyOn(translateTextService, 'init').and.callFake(
         (expId, languageCode, successCallback) => successCallback());
       broadcastSpy = spyOn(
         ckEditorCopyContentService, 'broadcastCopy').and.stub();
 
       component.ngOnInit();
-      target = document.createElement('div');
-      target.onclick = function(this, ev) {
+      nonParagraphTarget.onclick = function(this, ev) {
+        propagationSpy = spyOn(ev, 'stopPropagation').and.stub();
+        component.onContentClick(ev);
+      };
+      paragraphTarget.onclick = function(this, ev) {
         propagationSpy = spyOn(ev, 'stopPropagation').and.stub();
         component.onContentClick(ev);
       };
     }));
 
-    it('should broadcast the clicked element', () => {
-      target.click();
-      expect(broadcastSpy).toHaveBeenCalledWith(target);
+    it('should not broadcast the clicked paragraph element', () => {
+      paragraphTarget.click();
+      expect(broadcastSpy).not.toHaveBeenCalledWith(paragraphTarget);
+    });
+
+    it('should broadcast the clicked non paragraph element', () => {
+      nonParagraphTarget.click();
+      expect(broadcastSpy).toHaveBeenCalledWith(nonParagraphTarget);
+    });
+
+    it('should broadcast the clicked math element', () => {
+      paragraphTarget.append(mathTarget);
```

Comment:
As `mathTarget` is being appended to `paragraphTarget` here, `should not broadcast the clicked paragraph element` became flaky and failed some times when I test this in my local environment. That's why I thought of initialising `paragraphTarget` before each test. Your thoughts on this @sagangwee ?

Reply:
Ah, I see. It doesn't look like the variable will be torn down properly between tests. Then, it's fine to leave it. Thanks.

Reply:
Can we do that correctly? Re-initialize it in the beforeEach?